pkg/operation/botanist/component/metricsserver/metrics_server.go

```diff
@@ -62,32 +64,37 @@ const (
 func New(
 	client client.Client,
 	namespace string,
-	secretsManager secretsmanager.Interface,
-	image string,
-	vpaEnabled bool,
-	kubeAPIServerHost *string,
+	values Values,
 ) component.DeployWaiter {
 	return &metricsServer{
-		client:            client,
-		namespace:         namespace,
-		secretsManager:    secretsManager,
-		image:             image,
-		vpaEnabled:        vpaEnabled,
-		kubeAPIServerHost: kubeAPIServerHost,
+		client:    client,
+		namespace: namespace,
+		values:    values,
 	}
 }
 
+// Values is a set of configuration values for the metrics server component.
+type Values struct {
+	// Image is the container image used for the metrics server.
+	Image string
+	// SecretsManager is the manager responsible for managing metrics server's secrets.
+	SecretsManager secretsmanager.Interface
+	// VPAEnabled specifies is VPA should be enabled for the metrics server.
+	VPAEnabled bool
+	// KubeAPIServerHost is the kube-apiserver host name.
+	KubeAPIServerHost *string
+	// KubernetesVersion is the version of the for the Kubernetes components.
```

Comment:
Nit it should be rather: `KubernetesVersion is the version of the Kubernetes components.`

When we say `the version of the Kubernetes components` it is not clear. With the `WorkerPoolKubernetesVersion` feature the K8s version of the kubelet/kube-proxy can differ from the K8s version of the control plane components. I think here you are rather interested in the K8s version of the control plane.

That's why I would rather use:

```suggestion
	// KubernetesVersion is the Kubernetes version of the Shoot.
```

Can you also adapt the other occurrences of this doc string?